message center functions - fix formatting and writing, improve descriptions (#582)
vtexdocs/dev-portal-content: docs/guides/Message-Center/templates/how-to-set-up-functions-in-the-message-center-templates.md

updatedAt: "2022-02-10T20:58:11.681Z"
---

The purpose of this article is to present the most used commands and all the functions that can be used to boost your store’s email templates using Message Center.

To configure its templates, Message Center uses a language called Handlebars, with a few commands that make it easy to implement while expanding its customization.

> When editing the email template on VTEX Admin, in **Store Settings** > **Email Templates** > **Templates**, it is possible to use a range of variables provided by the system. To use these variables in the email layout, you must follow this Handlebars syntax: `{{class.attribute}}`. The `{{class.attribute}}` is obtained using the JSON Data field.

Reviewer: ```suggestion
When editing the email template on VTEX Admin, in **Store Settings** > **Email Templates** > **Templates**, it is possible to use a range of variables provided by the system. To use these variables in the email layout, you must follow this Handlebar's syntax: `{{class.attribute}}`. The `{{class.attribute}}` is obtained using the JSON Data field.
```

Author: O nome é [Handlebars](https://handlebarsjs.com/) mesmo, mas acho que dá para melhorar o texto aqui

Author: ```suggestion
When editing the email template on VTEX Admin, in **Store Settings** > **Email Templates** > **Templates**, it is possible to use a range of variables provided by the system. To use these variables in the email layout, you must follow this syntax using Handlebars: `{{class.attribute}}`. The `{{class.attribute}}` is obtained using the JSON Data field.
```

> >⚠️ The JSON attributes are the same for every VTEX store, however, the values may differ depending on the way you configured your store. So it is important to remember to configure your store relying on the info displayed by the [Get Order API](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/oms/pvt/orders/-orderId-) of your store.

Reviewer: ```suggestion
>⚠️ The JSON attributes are the same for every VTEX store, however, the values may differ depending on the way you configured your account. So it is important to remember to configure your store relying on the info displayed by the [Get Order API](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/oms/pvt/orders/-orderId-) of your store.
```

Reviewer: Achei que ficou um pouco repetitivo "store" no parágrafo e pouco preciso. O que pode mudar dependendo da configuração? Que configurações afetam?

Author: Não cheguei a mexer nesse callout, mas vale mudar ele todo 👀 Acho que é só dizer é que para configurar os templates de email transacional precisa se basear na get order api

Author: ```suggestion
>⚠️ Always configure transactional email templates relying on the information retrieved by the [Get Order API](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/oms/pvt/orders/-orderId-).
```

Check out the following example in which the template uses the `HostName` information from the JSON Data:

**HTML email template:**

```html
<strong>{{_accountInfo.HostName}}</strong>
```

**Example JSON Data:**

```json
"_accountInfo": { 
  "HostName": "lojadosuporte" 
}
```

**Result**: **lojadosuporte**

![exemplo html](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/images/how-to-set-up-functions-in-the-message-center-templates-0.png)

The above example can have different values when comparing Sellers and Marketplaces.

## Functions

The use of the functions follows the syntax `{{function class.attribute}}`.

Consider the following JSON Data example to understand the available functions, described in the table below.

**Example JSON Data:**

```json
"items": [ 
  { 

>     "name": "Produto A", 

Reviewer: ```suggestion
    "name": "Product A", 
```

Reviewer: Importante localizar exemplos também

    "sellingPrice": 20000, 
    "priceValidUntil": "2050-05-30T21:00:00Z", 
    "deliveryTime": "8bd" 
  }, 
  {

>     "name": "Produto B", 

Reviewer: ```suggestion
    "name": "Product B", 
```
